Oak NotImplementedError: `interp2d` has been removed in SciPy 1.14.0

### To reproduce
```
pytest forest/oak/tests
```
### Observed output
Current:
```
E       NotImplementedError: `interp2d` has been removed in SciPy 1.14.0.
E       
E       For legacy code, nearly bug-for-bug compatible replacements are
E       `RectBivariateSpline` on regular grids, and `bisplrep`/`bisplev` for
E       scattered 2D data.
E       
E       In new code, for regular grids use `RegularGridInterpolator` instead.
E       For scattered data, prefer `LinearNDInterpolator` or
E       `CloughTocher2DInterpolator`.
E       
E       For more details see
E       https://scipy.github.io/devdocs/tutorial/interpolate/interp_transition_guide.html
```
Original:
```
forest/oak/tests/test_identify_peaks_in_cwt.py::test_identify_peaks_in_cwt
  /Users/ils475/code/onnelalab/forest/forest/oak/base.py:177: DeprecationWarning: `interp2d` is deprecated!
  `interp2d` is deprecated in SciPy 1.10 and will be removed in SciPy 1.12.0.
  
  For legacy code, nearly bug-for-bug compatible replacements are
  `RectBivariateSpline` on regular grids, and `bisplrep`/`bisplev` for
  scattered 2D data.
  
  In new code, for regular grids use `RegularGridInterpolator` instead.
  For scattered data, prefer `LinearNDInterpolator` or
  `CloughTocher2DInterpolator`.
  
  For more details see
  `https://gist.github.com/ev-br/8544371b40f414b7eaf3fe6217209bff`
  
    ip = interpolate.interp2d(range(coefs.shape[1]), freqs, coefs)
```
### Expected output
no errors or warnings

### Reference
https://scipy.github.io/devdocs/tutorial/interpolate/interp_transition_guide.html

--- a/forest/oak/base.py
+++ b/forest/oak/base.py
@@ -184,8 +184,12 @@ def compute_interpolate_cwt(tapered_bout: np.ndarray, fs: int = 10,
     # interpolate coefficients
     freqs = out[2]
     freqs_interp = np.arange(0.5, 4.5, 0.05)
-    ip = interpolate.interp2d(range(coefs.shape[1]), freqs, coefs)
-    coefs_interp = ip(range(coefs.shape[1]), freqs_interp)
+    interpolator = interpolate.RegularGridInterpolator(
+        (freqs, range(coefs.shape[1])), coefs
+    )
+    grid_x, grid_y = np.meshgrid(freqs_interp, range(coefs.shape[1]),
+                                 indexing='ij')
+    coefs_interp = interpolator((grid_x, grid_y))
 
     # trim spectrogram from the coi
     coefs_interp = coefs_interp[:, 5*fs:-5*fs]

